src/createServer.js

```diff
@@ -1,11 +1,251 @@
 'use strict';
 
-// const express = require('express');
+const express = require('express');
+const cors = require('cors');
 
 function createServer() {
+  const app = express();
+
+  app.use(cors());
+
+  let users = [];
+  let expenses = [];
+
+  app.get('/expenses', express.json(), (req, res) => {
+    const {
+      userId,
+      categories,
+      from,
+      to,
+    } = req.query;
+
+    const filteredExpenses = expenses.filter(expense => {
+      const isUserIdMatch = userId
+        ? expense.userId === +userId
+        : true;
+
+      const isCategoriesMatch = categories
+        ? expense.category === categories
+        : true;
+
+      const isFromMatch = from
+        ? expense.spentAt >= from
+        : true;
+
+      const isToMatch = to
+        ? expense.spentAt <= to
+        : true;
+
+      return isUserIdMatch && isCategoriesMatch && isFromMatch && isToMatch;
+    });
+
+    res.send(filteredExpenses);
+  });
+
+  app.get('/expenses/:id', (req, res) => {
+    const { id } = req.params;
+
+    const foundExpense = expenses.find(expense => expense.id === +id);
+
+    if (!foundExpense) {
+      res.sendStatus(404);
+
+      return;
+    }
+
+    res.send(foundExpense);
+  });
+
+  app.post('/expenses', express.json(), (req, res) => {
+    const {
+      userId,
+      spentAt,
+      title,
+      amount,
+      category,
+      note,
+    } = req.body;
+
+    const requiredFields = ['userId', 'spentAt', 'title', 'amount', 'category'];
+
+    const requestFields = Object.keys(req.body);
+
+    const isReqFieldsValid
+      = requiredFields.every(field => requestFields.includes(field));
+
+    if (!isReqFieldsValid) {
+      res.sendStatus(400);
+
+      return;
+    }
+
+    const foundUser = users.find(user => user.id === +userId);
+
+    if (!foundUser) {
+      res.sendStatus(400);
+
+      return;
+    }
```

Comment:
why do you do it? it is an `add `method.

Reply:
these is a requirement in the tests, if the user is not found in the create new expense request, it shoud return status code 400
![Screenshot 2023-04-17 at 02 27 41](https://user-images.githubusercontent.com/57355180/232349314-58e19110-5aba-414e-934e-b7336aa7aec2.jpg)